Mage/src/main/java/mage/abilities/effects/common/ReturnSourceFromGraveyardToBattlefieldEffect.java

```diff
@@ -82,19 +89,28 @@ public boolean apply(Game game, Ability source) {
                     game.addEffect(effect, source);
                 }
             }
+            if (attacking) {
+                game.getCombat().addAttackingCreature(card.getId(), game);
+            }
         }
         return true;
     }
 
-    private void setText() {
+    @Override
+    public String getText(Mode mode) {
```

Comment:
Please change this to (and the WithCounter subclass) to set `staticText`. They do not need to be re-creating their text every time since it does not change.

Reply:
Do you mean the way it was before?  I thought overriding getText was the preferred way but maybe I'm wrong.  I see some classes doing this and some using setText().  I think we should be consistent one way or the other.

Reply:
Ok, makes sense.  I'll change that back.

Reply:
Done